client/app/scripts/superdesk/editor/editor.js

```diff
@@ -199,11 +203,125 @@ function FindReplaceCommand(rootNode) {
     }
 }
 
+function SpellcheckService($q) {
+
+    var selection = document.getSelection();
+
+    this.check = function check(text) {
+        var words = _.uniq(_.filter(text.split(/[\W0-9]+/)));
```

Comment:
I also suggest renaming the "words" variable, since it doesn't actually contain a list of words but a list of non-word sequences. On the first glance I thought you were extracting words from the text. ;)